### PR TITLE
SONAR-7125 fix temp dir cleaning

### PR DESCRIPTION
**Pull request by sns-seb**
fix it in App to avoid OOM if temp dir is big
fix it also in core, batch and server

### CODE REVIEW

File: server/sonar-process/src/main/java/org/sonar/process/FileUtils.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      throw new IOException(format("Directory '%s' is a file", directory.getAbsolutePath()));
+    }
+    deleteDirectoryImpl(path);
+
+    if (Files.exists(path)) {
+      throw new IOException(format("Unable to delete directory '%s'", directory.getAbsolutePath()));
+    }
+  }
+
+  private static Path toPath(File file) {
```
Comment (simonbrandhof) [inline server/sonar-process/src/main/java/org/sonar/process/FileUtils.java]: what's the value of such method ?

Comment (sns-seb) [inline server/sonar-process/src/main/java/org/sonar/process/FileUtils.java]: I initially used `Path.get(file.toURI())`, I'll remove it

**Comment by simonbrandhof** [comment]
We should find a way to not duplicate classes between different processes. What about implementing later a module "sonar-utils", which will basically replace "sonar-core" ?
Else LGTM.

**Comment by sns-seb** [comment]
PR updated with fix for cleaning directory through sym link

File: sonar-core/src/main/java/org/sonar/core/util/FileUtils.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    try {
+      if (file.isDirectory()) {
+        deleteDirectory(file);
+      } else {
+        Files.delete(file.toPath());
+      }
+      return true;
+    } catch (IOException | SecurityException ignored) {
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/util/FileUtils.java]: CRITICAL Either log or rethrow this exception. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)

File: server/sonar-process/src/main/java/org/sonar/process/FileUtils.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    try {
+      if (file.isDirectory()) {
+        deleteDirectory(file);
+      } else {
+        Files.delete(file.toPath());
+      }
+      return true;
+    } catch (IOException | SecurityException ignored) {
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/FileUtils.java]: CRITICAL Either log or rethrow this exception. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* CRITICAL 2 critical
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**